node-graph/gcore/src/vector/generator_nodes.rs

```diff
@@ -6,19 +6,96 @@ use bezier_rs::Subpath;
 
 use glam::DVec2;
 
-pub struct UnitCircleGenerator;
+#[derive(Debug, Clone, Copy)]
+pub struct UnitCircleGenerator<Radius> {
+	radius: Radius,
+}
 
 #[node_macro::node_fn(UnitCircleGenerator)]
-fn unit_circle(_input: ()) -> VectorData {
-	super::VectorData::from_subpath(Subpath::new_ellipse(DVec2::ZERO, DVec2::ONE))
+fn unit_circle(_input: (), radius: f32) -> VectorData {
+	let radius = radius.into();
+	super::VectorData::from_subpath(Subpath::new_ellipse(DVec2::splat(radius * -1.), DVec2::splat(radius)))
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct UnitElipseGenerator<RadiusX, RadiusY> {
+	radius_x: RadiusX,
+	radius_y: RadiusY,
+}
+
+#[node_macro::node_fn(UnitElipseGenerator)]
+fn unit_elipse(_input: (), radius_x: f32, radius_y: f32) -> VectorData {
+	let radius_x = radius_x.into();
+	let radius_y: f64 = radius_y.into();
+	let corner1 = DVec2::new(radius_x * -1., radius_y * -1.);
+	let corner2: DVec2 = DVec2::new(radius_x, radius_y);
+	super::VectorData::from_subpath(Subpath::new_ellipse(corner1, corner2))
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct UnitRectangleGenerator<SizeX, SizeY> {
+	size_x: SizeX,
+	size_y: SizeY,
+}
+
+#[node_macro::node_fn(UnitRectangleGenerator)]
+fn unit_square(_input: (), size_x: f32, size_y: f32) -> VectorData {
+	let size_x = (size_x / 2.).into();
+	let size_y = (size_y / 2.).into();
+	let corner1 = DVec2::new(size_x * -1., size_y * -1.);
+	let corner2 = DVec2::new(size_x, size_y);
```

Comment:
The size here is actual half of the side - either rename it or change it to be the actual size:
```rs
let size = DVec2::new(size_x as f64, size_y as f64);
let corner1 = -size / 2.;
let corner2 = size / 2.;
```